android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/HelpFile.java

```diff
@@ -36,15 +40,16 @@
   public static Intent toActionView(Context context, String helpFile, String packageID) {
     Intent i = new Intent(Intent.ACTION_VIEW);
 
-    if (FileUtils.isWelcomeFile(helpFile) && ! KMManager.isTestMode()) {
+    if ((FileUtils.isWelcomeFile(helpFile) || FileUtils.hasPDFExtension(helpFile))
+        && ! KMManager.isTestMode()) {
       File customHelp = new File(new File(helpFile).getAbsolutePath());
       i.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
       // Starting with Android N, you can't pass file:// to intents, so we use FileProvider
       try {
         final String authority = FileProviderUtils.getAuthority(context);
         Uri contentUri = FileProvider.getUriForFile(
           context, authority, customHelp);
-        i.setDataAndType(contentUri, "text/html");
+        i.setDataAndType(contentUri, "application/pdf");
```

Comment:
Will this set the type to `application/pdf` for `welcome.htm` as well as `.pdf` files?

Reply:
With the changes on this PR, html files are handled by KMHelpFileActivity, while only pdf files are handled by HelpFile.

I'll clean HelpFile to make this clear.